src/Core/Localization/Currency/CurrencyDataSource.php

```diff
@@ -88,17 +82,29 @@ public function isCurrencyAvailable($currencyCode)
     }
 
     /**
-     * Get all the available (installed + active) currencies' data.
-     *
+     * {@inheritdoc}
+     */
+    public function getAvailableCurrenciesData($localeCode)
+    {
+        return $this->formatCurrenciesData($this->installedDataLayer->getAvailableCurrencyCodes(), $localeCode);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getAllCurrenciesData($localeCode)
+    {
+        return $this->formatCurrenciesData($this->installedDataLayer->getAvailableCurrencyCodes(), $localeCode);
+    }
+
+    /**
+     * @param array $currencyCodes
      * @param string $localeCode
-     *                           IETF tag. Data will be translated in this language
      *
-     * @return CurrencyData[]
-     *                        The available currencies' data
+     * @return array
      */
-    public function getAvailableCurrenciesData($localeCode)
+    protected function formatCurrenciesData(array $currencyCodes, $localeCode)
```

Comment:
Since class is not `final`, I'd rather make this method `private` (unless there is a good reason not to), otherwise it's part of public interface.